internal/webhooks/cluster_test.go

```diff
@@ -1004,7 +1004,8 @@ func TestClusterValidation(t *testing.T) {
 			// Create the webhook.
 			webhook := &Cluster{}
 
-			_, err := webhook.validate(ctx, tt.old, tt.in)
+			warnings, err := webhook.validate(ctx, tt.old, tt.in)
+			g.Expect(warnings).To(BeNil())
```

Comment:
nit: I wonder if we should check for empty instead of nil. Just because it's slightly more high-level. We don't really care about if we get nil or an empty slice (that's an implementation detail). We only care that we don't get a warning.